### PR TITLE
You can't withdraw credits from budget card anymore

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
* Budget card is no longer withdraw-able
* Budget card can be emagged for withdraw feature unlock
(space pirates just can bypass this lock)
* budget to budget transfer feature - touch your budget card to the target budget card

## Why It's Good For The Game
extreme amount of money shouldn't be that easily accessible, but also heads should utilise their console to give payment bonus, not just withdraw money from the budget card when they want to give bonus.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

(Image missing somehow. It tells a warning instead of allowing you to withdraw)

You can't withdraw

image

image

emag can make it withdrew

---------------

image

3571 to both

----

image

when you touch a budget card with a budget card

----

image
image
image

after transferred

</details>

## Changelog
:cl:
balance: budget card is locked and you can't withdraw credits from it. You can withdraw when you emag the cards, or be a space pirate to withdraw freely.
add: budget card can transfer its money into another budget card when you touch it into another card.
/:cl:

### CODE REVIEW

**Comment by Dejaku51** [comment]
now i can't ask other departaments to loan money to make station goal :tragedy:

**Comment by EvilDragonfiend** [comment]
> now i can't ask other departaments to loan money to make station goal :tragedy:

hmm, maybe budget-budget transfer feature

**Comment by EvilDragonfiend** [comment]
This PR should be merged with https://github.com/BeeStation/BeeStation-Hornet/pull/7559 together

File: code/game/objects/items/cards_ids.dm [hunk 1]
```diff
@@ -760,6 +760,46 @@ update_label("John Doe", "Clowny")
...
+/obj/item/card/id/departmental_budget/AltClick(mob/living/user)
+	if(!alt_click_can_use_id(user))
+		return
+
+	var/datum/antagonist/pirate/P = user.mind.has_antag_datum(/datum/antagonist/pirate)
+	if(!((obj_flags & EMAGGED) || P)) // snowflake. pirates can just bypass.
+		to_chat(user, "<span class='warning'>Budget card isn't allowed to withdraw credits directly.</span>")
+		return
+
+	..()
```
Comment (Archanial) [inline code/game/objects/items/cards_ids.dm:772]:
```suggestion
	return ..()
```

File: code/game/objects/items/cards_ids.dm [hunk 2]
```diff
@@ -760,6 +760,46 @@ update_label("John Doe", "Clowny")
...
+			return
+		if(B_from.adjust_money(-amount_to_remove))
+			registered_account.adjust_money(amount_to_remove)
+			to_chat(user, "<span class='notice'>You transfered [amount_to_remove] credits into the budget card.</span>")
+			return
+		else
+			var/difference = amount_to_remove - B_from.account_balance
+			B_from.bank_card_talk("<span class='warning'>ERROR: The linked account requires [difference] more credit\s to perform that transfer.</span>", TRUE)
+			return
+	else
```
Comment (Archanial) [inline code/game/objects/items/cards_ids.dm]: redundant else

File: code/game/objects/items/cards_ids.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/game/objects/items/cards_ids.dm:772]: Does alt click return?

Comment (EvilDragonfiend) [reply code/game/objects/items/cards_ids.dm:772]: It doesn't return any value though, but okay.

Comment (PowerfulBacon) [reply code/game/objects/items/cards_ids.dm:772]: Don't return then

**Comment by PowerfulBacon** [comment]
This is interesting and I can think of individual, very specific situations for and against this, but what's the main intent of this PR? What do you want to change as a result of this change?

**Comment by EvilDragonfiend** [comment]
> This is interesting and I can think of individual, very specific situations for and against this, but what's the main intent of this PR? What do you want to change as a result of this change?

Reasons:
* For heads, they'll be no longer allowed to directly withdraw the budget money. If they want to give bonus to their underlings, they'll have to use their payment console to give. (main reason)
* But also, if they want to use money directly, they'll still have to use a payment console for giving a lot of bonus to themself.
* If a person wants to steal credits in a budget card, they'll have to steal the whole card instead of just pulling out all credits. (or emag it) so, a thief will have to keep the budget card all day if they want to abuse it.
* and last, I am writing a code to track how the credits go. I don't like a case that withdraws credits from a budget card.

**Comment by itsmeow** [comment]
FINANCE LOGS FOR THE LOG VIEWER ?!?!

**Comment by EvilDragonfiend** [comment]
> FINANCE LOGS FOR THE LOG VIEWER ?!?!

I am trying to add "INVESTIGATE_ECONOMY_CREDITS" to investigate log.

**Comment by NormalAnomaly** [comment]
I am still unclear at the actual intent and practical implications of this considering most of my gameplay experience suggests that most of the cases money is taken out of a budget is not malicious.

Heads have never made strong use of their console, ever, in any capacity.  To the point that I've met more than one that had no idea they had it.  Even among those that do know its there and know its function, for job transfers "Go see the hop" is still the usual answer.

Heads very rarely ever give bonuses, with console or directly, or even hand out the medals.  The idea of giving a financial reward is the sort of thing that might happen a handful of times a week, I'd wager easily below 5% of shifts.  Making it take more steps will not benefit this.  Heads that actually reward workers with more than "Cool" or "Good Job" are not the norm by far.

Thieves do steal the whole budget card pretty often, in part to abuse its access for private ordered goods if they plan to even do anything with the money.  If they do leave the card, then the card will continue to gain funds over the course of the shift and the rightful user might still make use of it later.  And, as a corner case example, I've stopped teslagriefers ordering parts or funnymen ordering 30 snake crates by withdrawing credits.  The QM should at least retain the ability to freeze transactions.

"extreme amount of money shouldn't be that easily accessible"  Why and what difference does it make?
Stealing head lockers is arguably too easy, though the budget should probably be carried by the head so its an IC issue to a degree that it gets stolen out of the lockers if there is an active head or a problem with command gameplay/population if the head slot is empty.  Its enough cash to buy a few crates, is that an extreme amount (cargo budget aside)?  Cash has no practical value other than cargo orders, and in my QM time, most of the stolen budget funds have just been used by a clown to try to get a gimmick crate.  And since the cargo consoles just take directly from the cargo budget, anyone who breaks into cargo functionally has access to the cargo budget.

**Review by Archanial — CHANGES_REQUESTED** [review]
I want to see some method traitors can use to withdraw money, that doesn't rely on emag alone, but in return takes more time (perhaps acid?).

**Comment by EvilDragonfiend** [comment]
> I want to see some method traitors can use to withdraw money, that doesn't rely on emag alone, but in return takes more time (perhaps acid?).

I will think about it.

**Comment by EvilDragonfiend** [comment]
this PR should be merged after the PR I linked here. It will be set to draft until the primary PR is merged.

**Comment by Kenionatus** [comment]
Disclaimer: not a Bee player, instead playing on NSV (Bee downstream)
I would argue that heads and other non-antag crew members having direct access to the money and being able to embezzle it is actually a good thing. It gives one more opportunity for IC conflict. As @NormalAnomaly said: clown buying meme crate with a department's money.
Another downside of this PR is that buying from a budget card increases the cost in the same way that buying from a personal budget does, as far as I can tell (unless the interface is lying). Even if transferring to the cargo budget through the head's console is possible that would be another hoop to jump through, which imo doesn't really add significant gameplay.

**Comment by EvilDragonfiend** [comment]
I will not cancel this basic idea, but I will do that when the game can cover things a lot. The game needs more mechanics to support this feature.
I am not against embezzlement, but I don't like how easy it is.

**Pull request closed without merging**